### PR TITLE
feat: introduce checkbox variants placeholders

### PR DESCRIPTION
**Pull request by devodii**

### CODE REVIEW

File: libs/ui/design-system/src/lib/checkbox.tsx [hunk 1]
```diff
@@ -1,17 +1,32 @@
...
 import { Label } from '@nmit-coursition/ui/primitives/label'
+import { cn } from '@nmit-coursition/ui/utils'
+import { type VariantProps, cva } from 'class-variance-authority'
 
-interface CheckboxProps {
+const checkboxVariants = cva('', {
+  variants: {
+    variant: {
+      outlined: '',
+      critical: '', // applies a red border
```
Comment (BleedingDev) [inline libs/ui/design-system/src/lib/checkbox.tsx:10]: **suggestion:** Critical is not a status, **red** colour is applied when checkbox is invalid (e.g. it has to be checked like _"I agree with Terms & Conditions."_).

File: libs/ui/design-system/src/lib/checkbox.tsx [hunk 2]
```diff
@@ -1,17 +1,32 @@
...
+    },
+    size: {
+      sm: '', // 8px
```
Comment (BleedingDev) [inline libs/ui/design-system/src/lib/checkbox.tsx:13]: **question:** Where to we have different sizes defined in Figma? We have only one-size of checkbox, right?

**Review by BleedingDev — CHANGES_REQUESTED** [review]

File: libs/ui/design-system/src/lib/checkbox.tsx (hunk 2 above)
Comment (devodii) [reply libs/ui/design-system/src/lib/checkbox.tsx:13]: I actually saw 2 sizes when i inspected, that was why i specified in the comments.
But, I'm going to remove all the unnecessary variants as i already did in other components.

File: libs/ui/design-system/src/lib/checkbox.tsx [hunk 3]
```diff
@@ -1,17 +1,31 @@
...
 import { Label } from '@nmit-coursition/ui/primitives/label'
+import { cn } from '@nmit-coursition/ui/utils'
+import { type VariantProps, cva } from 'class-variance-authority'
 
-interface CheckboxProps {
+const checkboxVariants = cva('', {
+  variants: {
+    variant: {
+      primary: '',
+      outlined: '',
```
Comment (BleedingDev) [inline libs/ui/design-system/src/lib/checkbox.tsx:10]: **suggestion:** Same thing as before, there's no `outlined` variant, so `Checkbox` probably has no variants at all. It's `focused` state.

File: libs/ui/primitives/src/checkbox.tsx [hunk 4]
```diff
@@ -4,23 +4,25 @@ import * as React from 'react'
...
-    <CheckboxPrimitive.Indicator className={cn('flex items-center justify-center text-current')}>
-      <Check className='h-4 w-4' />
-    </CheckboxPrimitive.Indicator>
-  </CheckboxPrimitive.Root>
-))
-Checkbox.displayName = CheckboxPrimitive.Root.displayName
+export interface RootProps extends React.ComponentPropsWithoutRef<typeof CheckboxPrimitive.Root> {}
 
-export { Checkbox }
+const Main = React.forwardRef<React.ElementRef<typeof CheckboxPrimitive.Root>, RootProps>(
```
Comment (BleedingDev) [inline libs/ui/primitives/src/checkbox.tsx:9]: **suggestion:** We shouldn't mutate generated code.

**Review by BleedingDev — CHANGES_REQUESTED** [review]

File: libs/ui/primitives/src/checkbox.tsx (hunk 4 above)
Comment (devodii) [reply libs/ui/primitives/src/checkbox.tsx:9]: I reverted everything back to how Shadcn originally ships it, but I added an export for the primitive root component's type to allow extension within the design system.

**Pull request closed without merging**